homeassistant/components/dwd_weather_warnings/config_flow.py

```diff
@@ -25,27 +29,72 @@ async def async_step_user(
         errors: dict = {}
 
         if user_input is not None:
-            region_identifier = user_input[CONF_REGION_IDENTIFIER]
+            valid_options = (CONF_REGION_IDENTIFIER, CONF_REGION_DEVICE_TRACKER)
 
-            # Validate region identifier using the API
-            if not await self.hass.async_add_executor_job(
-                DwdWeatherWarningsAPI, region_identifier
-            ):
-                errors["base"] = "invalid_identifier"
+            # Check, if either CONF_REGION_IDENTIFIER or CONF_GPS_TRACKER has been set.
+            if all(k not in user_input for k in valid_options):
+                errors["base"] = "no_identifier"
+            elif all(k in user_input for k in valid_options):
+                errors["base"] = "ambiguous_identifier"
```

Comment:
This seems a bit unnecessary. Maybe just do a set intersection and check the length?
Not sure if my suggestion is more readable though.
```py
EXCLUSIVE_OPTIONS = {CONF_REGION_IDENTIFIER, CONF_REGION_DEVICE_TRACKER}
options = user_input.keys() & EXCLUSIVE_OPTIONS
if not options:
    ...
elif len(options) == 2:
   ...
```

Reply:
In my opinion the current solution is easier to understand though not as efficient probably

Reply:
That's fine, it's a matter of taste 👍